### PR TITLE
update: Update to Twitch4J 1.22.0

### PR DESCRIPTION
**Pull request by Awakened-Redstone**
### Prerequisites for Code Changes

* [x] This pull request follows the code style of the project

### Changes Proposed

* Updates dependencies to match **Twitch4J 1.22.0**
* Updates to **Twitch4J 1.22.0**
* Bumps mod version to `1.8.0` as the **Twitch4J** version changed

### CODE REVIEW

**Review by iProdigy — APPROVED** [review]
thank you!

**Review by iProdigy — APPROVED** [review]

**Review by PhilippHeuer — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - name: Set up JDK 21
+        uses: actions/setup-java@2dfa2011c5b2a0f1489bf9e433881c92c1631f88 # v4.3.0
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
```
Comment (iProdigy) [inline .github/workflows/release.yml]:
```suggestion
      - uses: gradle/actions/setup-gradle@d156388eb19639ec20ade50009f3d199ce1e2808 # v4.1.0
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Publish mod
+
+on:
+  push:
+    branches: [ "main" ]
+
+permissions:
+  contents: write
```
Comment (iProdigy) [inline .github/workflows/release.yml]: can this be `read` instead?

Comment (Awakened-Redstone) [reply .github/workflows/release.yml]: oh, likely, I used a template that allowed publishing to github pages too

**Review by iProdigy — APPROVED** [review]

**Pull request merged**